### PR TITLE
Fixing the platform warning and  bad URI(is not URI?) error

### PR DESCRIPTION
**Pull request by Ajaykn21**
# Issue: 1
As part of the config key :file, we are checking the supported file extensions based on environment variable `SharedValues::PLATFORM_NAME`.
And if this is not present, we are showing incorrect error message.

`[19:34:18]: Unknown platform '', consider using one of: [:android, :ios, :mac, :windows, :custom]`

This is slightly confusing when users are using this for only upload action and doesn't want to set the platform.

# Issue 2
When user gives AppName with spaces/special characters, then url construction for GET is failing with **URI::InvalidURIError: [!] bad URI(is not URI?): v0.1/apps/korea/test app**
If we do URI.encode this error is resolved but fails in app_creation steps with validation errors.

Hence handling this just by rescue-ing on URI::InvalidURIError.

### CODE REVIEW

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb [hunk 1]
```diff
@@ -233,8 +233,13 @@ def self.get_or_create_app(params)
...
         }
 
-        if Helper::AppcenterHelper.get_app(api_token, owner_name, app_name)
-          return true
+        begin
+          if Helper::AppcenterHelper.get_app(api_token, owner_name, app_name)
+            return true
+          end
+        rescue URI::InvalidURIError
+          UI.user_error!("Provided app_name: '#{app_name}' is not in a valid format. Please ensure no special charecters and spaces in the app_name.")
```
Comment (jpeg-ms) [inline lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb]: Typo here: "characters or spaces"

File: lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb [hunk 2]
```diff
@@ -407,13 +412,15 @@ def self.available_options
...
                               verify_block: proc do |value|
                                 platform = Actions.lane_context[SharedValues::PLATFORM_NAME]
-                                accepted_formats = Constants::SUPPORTED_EXTENSIONS[platform.to_sym] if platform
-                                unless accepted_formats
-                                  UI.important("Unknown platform '#{platform}', consider using one of: #{Constants::SUPPORTED_EXTENSIONS.keys}")
-                                  accepted_formats = Constants::ALL_SUPPORTED_EXTENSIONS
+                                if platform
+                                  accepted_formats = Constants::SUPPORTED_EXTENSIONS[platform.to_sym]
+                                  unless accepted_formats
+                                    UI.important("Unknown platform '#{platform}', Supported are #{Constants::SUPPORTED_EXTENSIONS.keys}")
```
Comment (jpeg-ms) [inline lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb]: Why change the text? Maybe rephrase as "Supported platforms: "

Comment (Ajaykn21) [reply lib/fastlane/plugin/appcenter/actions/appcenter_upload_action.rb]: Original text suggests user to set which is confusing: `consider using one of:`.. hence changed

**Comment by jpeg-ms** [comment]
This is to fix #152
